pkg/operator/admission/validation/pullsecret/secret_utils.go

```diff
@@ -0,0 +1,125 @@
+package pullsecret
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type authsStruct struct {
+	Auths map[string]Auth `json:"auths"`
+}
+
+type Auth struct {
+	Auth string `json:"auth"`
+}
+
+func unmarshalRequestToSecret(request *admissionv1.AdmissionRequest) (corev1.Secret, corev1.Secret, error) {
+	secret := corev1.Secret{}
+	oldSecret := corev1.Secret{}
+	var err error
+	//new object not defined on deletion
+	if request.Operation != admissionv1.Delete {
+		err = json.Unmarshal(request.Object.Raw, &secret)
+
+		if err != nil {
+			return corev1.Secret{}, corev1.Secret{}, err
+		}
+	}
+
+	if request.Operation == admissionv1.Update || request.Operation == admissionv1.Delete {
+		err = json.Unmarshal(request.OldObject.Raw, &oldSecret)
+	}
+	return secret, oldSecret, err
+}
+
+func secretIsOCM(new, old authsStruct) bool {
+	//pullsecret is for ICM only if it has the "cloud.openshift.com" cred
```

Comment:
OCM

Reply:
I'll fix it once we get approval from leads, don't want to dismiss reviews from other engineers yet